File: java/src/main/java/com/allstask/models/InvoiceStatus.java
// This file is @generated
package com.allstask.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.allstask.Utils.ToQueryParam;

public enum InvoiceStatus implements ToQueryParam {
    DRAFT("DRAFT"),
    FINALIZED("FINALIZED"),
    UNCOLLECTIBLE("UNCOLLECTIBLE"),
    VOID("VOID");
    private final String value;

    InvoiceStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: typescript/src/models/invoiceStatus.ts
// this file is @generated
/* eslint @typescript-eslint/no-explicit-any: 0 */

export enum InvoiceStatus {
    Draft = 'DRAFT',
    Finalized = 'FINALIZED',
    Uncollectible = 'UNCOLLECTIBLE',
    Void = 'VOID',
    }

export const InvoiceStatusSerializer = {
    _fromJsonObject(object: any): InvoiceStatus {
        return object;
    },

    _toJsonObject(self: InvoiceStatus): any {
        return self;
    }
}